parsec/core/pki/submitter.py

```diff
@@ -118,47 +128,47 @@ async def submit(
             submit_payload=raw_submit_payload,
         )
 
-        if rep["status"] == "already_submitted":
-            submitted_on = rep["submitted_on"]
+        if isinstance(rep, PkiEnrollmentSubmitRepAlreadySubmitted):
+            submitted_on = rep.submitted_on
             raise PkiEnrollmentSubmitCertificateAlreadySubmittedError(
                 f"The certificate `{self.x509_certificate.certificate_sha1.hex()}` has already been submitted on {submitted_on}",
                 rep,
                 self.enrollment_id,
                 self.x509_certificate,
             )
 
-        if rep["status"] == "enrollment_id_already_used":
+        if isinstance(rep, PkiEnrollmentSubmitRepIdAlreadyUsed):
             raise PkiEnrollmentSubmitEnrollmentIdAlreadyUsedError(
                 f"The enrollment ID {self.enrollment_id.hex} is already used",
                 rep,
                 self.enrollment_id,
                 self.x509_certificate,
             )
 
-        if rep["status"] == "already_enrolled":
+        if isinstance(rep, PkiEnrollmentSubmitRepAlreadyEnrolled):
             raise PkiEnrollmentSubmitCertificateAlreadyEnrolledError(
                 f"The certificate `{self.x509_certificate.certificate_sha1.hex()}` has already been enrolled",
                 rep,
                 self.enrollment_id,
                 self.x509_certificate,
             )
 
-        if rep["status"] == "email_already_used":
+        if isinstance(rep, PkiEnrollmentSubmitRepEmailAlreadyUsed):
             raise PkiEnrollmentSubmitCertificateEmailAlreadyUsedError(
                 f"The email address `{self.x509_certificate.subject_email_address}` is already used by an active user",
                 rep,
                 self.enrollment_id,
                 self.x509_certificate,
             )
 
-        if rep["status"] != "ok":
+        if not isinstance(rep, PkiEnrollmentSubmitRepOk):
             raise PkiEnrollmentSubmitError(
-                f"Backend refused to create enrollment: {rep['status']}",
+                f"Backend refused to create enrollment: {type(rep)}",
```

Comment:
What the expected string here ?

On my side, I get 

```python
>>> from parsec._parsec import PkiEnrollmentAcceptRepInvalidPayloadData
>>> print(type(PkiEnrollmentAcceptRepInvalidPayloadData("foo")))
<class 'builtins.PkiEnrollmentAcceptRepInvalidPayloadData'>
```

don't we want something like ?

```python
>>> print(type(PkiEnrollmentAcceptRepInvalidPayloadData("foo")).__name__)
PkiEnrollmentAcceptRepInvalidPayloadData
```